osu.Game/Rulesets/UI/DrawableRulesetDependencies.cs

```diff
@@ -68,7 +59,7 @@ public DrawableRulesetDependencies(Ruleset ruleset, IReadOnlyDependencyContainer
             SampleStore.PlaybackConcurrency = OsuGameBase.SAMPLE_CONCURRENCY;
             CacheAs(SampleStore = new FallbackSampleStore(SampleStore, parent.Get<ISampleStore>()));
 
-            CacheAs(ShaderManager = new RulesetShaderManager(host.Renderer, new NamespacedResourceStore<byte[]>(resources, @"Shaders"), parent.Get<ShaderManager>()));
+            parent.Get<ShaderManager>().AddStore(ruleset.ShortName, new NamespacedResourceStore<byte[]>(resources, @"Shaders"));
```

Comment:
I'm not really fond of this approach, it feels more like a band-aid than a proper solution. Keeping the shader store alive after `DrawableRulesetDependencies` is disposed sounds pretty bad when it's the one who added the store in the first place.

This should be done at a high level during game startup (e.g. `OsuGame`/`OsuGameBase`), but either way making ruleset shaders available in non-gameplay scenes sounds bad generally. 

I would prefer something along the lines of a `RulesetShaderCache` that creates and caches `ShaderManager`s for each ruleset throughout the game's lifetime and then have `DrawableRulesetDependencies` cache the ruleset's shader manager and add the main shader manager as a fallback store (with framework support for `AddStore(ShaderManager)`)